src/sinks/aws_s3/tests.rs

```diff
@@ -220,7 +220,7 @@ mod integration_tests {
 
         let (_lines, events, receiver) = make_events_batch(1, 1);
         sink.run(events).await.unwrap();
-        assert_eq!(receiver.await, BatchStatus::Errored);
+        assert_eq!(receiver.await, BatchStatus::Failed);
```

Comment:
It's not clear to me why this switch would happen, but I'm not opposed. Mostly I want to flag that there's some logic behind this change, which I am missing. I have to look up the difference between 'failed' and 'errored' every time; they are "at least one event in the batch had a permanent failure" and "at least one event in the batch had a transient error in delivery" respectively. These definitions lead to tricky questions about priority if a batch has an event that suffered a permanent failure and also a transient error, so I would tend to imagine the line between them is blurry in practice in vector today.